File: src/main/java/no/nav/bidrag/beregn/nettobarnetilsyn/beregning/NettoBarnetilsynBeregning.java
package no.nav.bidrag.beregn.nettobarnetilsyn.beregning;

import java.util.List;
import no.nav.bidrag.beregn.nettobarnetilsyn.bo.GrunnlagBeregning;
import no.nav.bidrag.beregn.nettobarnetilsyn.bo.ResultatBeregning;

public interface NettoBarnetilsynBeregning {

  List<ResultatBeregning> beregn(GrunnlagBeregning grunnlagBeregning);

  static NettoBarnetilsynBeregning getInstance(){
    return new NettoBarnetilsynBeregningImpl();
  }
}

File: src/main/kotlin/no/nav/bidrag/beregn/nettobarnetilsyn/beregning/NettoBarnetilsynBeregning.kt
package no.nav.bidrag.beregn.nettobarnetilsyn.beregning

import no.nav.bidrag.beregn.nettobarnetilsyn.bo.GrunnlagBeregning
import no.nav.bidrag.beregn.nettobarnetilsyn.bo.ResultatBeregning

fun interface NettoBarnetilsynBeregning {
    fun beregn(grunnlag: GrunnlagBeregning): List<ResultatBeregning>

    companion object {
        fun getInstance(): NettoBarnetilsynBeregning {
            return NettoBarnetilsynBeregningImpl()
        }
    }
}
